pkg/issuer/acme/dns/dns.go

```diff
@@ -415,9 +416,9 @@ func (s *Solver) prepareChallengeRequest(issuer v1alpha1.GenericIssuer, ch *v1al
 
 var errNotFound = fmt.Errorf("failed to determine DNS01 solver type")
 
-func (s *Solver) dns01SolverForConfig(config *v1alpha1.ACMEChallengeSolverDNS01) (webhook.Solver, interface{}, error) {
+func (s *Solver) dns01SolverForConfig(config *v1alpha1.ACMEChallengeSolverDNS01) (webhook.Solver, *v1alpha1.ACMEIssuerDNS01ProviderWebhook, error) {
 	solverName := ""
-	var c interface{}
+	var c *v1alpha1.ACMEIssuerDNS01ProviderWebhook
```

Comment:
So this should actually still be `interface{}` as we will have additional types of solver types that can be returned here.

I'll re-address this in a follow up PR, as I've just noticed that the RFC2136 provider should also have an entry here and doesn't 🙈